### PR TITLE
fix: rotate class id when branch more than 8

### LINKED ISSUE #3149: Git graph with more than 8 branches are not working properly
## Reproduce steps

Use the snippet [here](https://github.com/cilerler/cilerler.github.io/wiki/Branches#graph-)

## Expected

Like other branches, commits should be presented correctly.

## Actual

As seen in 1..5, the branch-off from the `develop` is not shown correctly.

image

### PR DESCRIPTION
**Pull request by faisalarbain**
## :bookmark_tabs: Summary
Arrow are not visible when total branches more than 8. The issue also reported in ticket below:
Resolves #3149

## :straight_ruler: Design Decisions
The fix is following current implementation for branch tag's color under `drawBranches` methods where it adjustIndexForTheme by deducting with 8 (magic number)

I extract the magic number into const to make it easy to refactor/change the value in future.

## Screenshot
This is using snippet provided in #3149 

SCR-20220618-p0r

### CODE REVIEW

**Comment by faisalarbain** [comment]
i not sure why static analysis is failing. can someone check?

File: src/diagrams/git/gitGraphRenderer.js [hunk 1]
```diff
@@ -185,15 +187,23 @@ const drawCommits = (svg, commits, modifyGraph) => {
...
           circle2.attr('r', 6);
           circle2.attr(
             'class',
-            'commit ' + typeClass + ' ' + commit.id + ' commit' + branchPos[commit.branch].index
+            'commit ' +
+              typeClass +
+              ' ' +
+              commit.id +
+              ' commit' +
+              (branchPos[commit.branch].index % THEME_COLOR_LIMIT)
```
Comment (sidharthv96) [inline src/diagrams/git/gitGraphRenderer.js:206]: nit.

```suggestion
            `commit ${typeClass} ${commit.id} commit${branchPos[commit.branch].index % THEME_COLOR_LIMIT}`
```
Same with other instances below.

File: src/diagrams/git/gitGraphRenderer.js [hunk 2]
```diff
@@ -462,7 +477,7 @@ const drawBranches = (svg, branches) => {
   const gitGraphConfig = getConfig().gitGraph;
   const g = svg.append('g');
   branches.forEach((branch, index) => {
-    let adjustIndexForTheme = index >= 8 ? index - 8 : index;
+    let adjustIndexForTheme = index % THEME_COLOR_LIMIT;
```
Comment (sidharthv96) [inline src/diagrams/git/gitGraphRenderer.js:480]:
```suggestion
    const adjustIndexForTheme = index % THEME_COLOR_LIMIT;
```

**Comment by sidharthv96** [comment]
@faisalarbain please add a snapshot test in `cypress/e2e/rendering/gitGraph.spec.js`

**Comment by faisalarbain** [comment]
@sidharthv96 added and rebased

**Review by Yash-Singh1 — APPROVED** [review]

**Comment by mclate** [comment]
Is there any roadmap or eta on when this one can be merged?

**Comment by faisalarbain** [comment]
rebased to resolve conflict. but some test were break not related to the changes

**Comment by knsv** [comment]
Thanks @faisalarbain! Merging!

**Pull request merged**